egs/swbd/s5c/local/chain/multi_condition/run_tdnn_7b.sh

```diff
@@ -0,0 +1,234 @@
+# smoothing options
+pool_window=
+pool_type='none'
+pool_lpfilter_width=
+self_repair_scale=0.00001
+# training options
+num_epochs=2
+initial_effective_lrate=0.001
+final_effective_lrate=0.0001
+leftmost_questions_truncate=-1
+max_param_change=2.0
+final_layer_normalize_target=0.5
+num_jobs_initial=3
+num_jobs_final=16
+minibatch_size=128
+relu_dim=625
+frames_per_eg=150
+remove_egs=true
+common_egs_dir=
+xent_regularize=0.1
+
+
+
+# End configuration section.
+echo "$0 $@"  # Print the command line for logging
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+# The iVector-extraction and feature-dumping parts are the same as the standard
+# nnet3 setup, and you can skip them by setting "--stage 8" if you have already
+# run those things.
+
+
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+
+dir=${dir}${affix:+_$affix}${suffix}_rvb${num_data_reps}
+train_set=${clean_train_set}${suffix}_rvb${num_data_reps}
+lang=data/lang_chain_2y
+treedir=exp/chain/tri5_2y_tree${suffix}
+lat_dir=exp/tri4_lats_nodup${suffix}
+rvb_lat_dir=${lat_dir}_rvb${num_data_reps}
+
+
+# if we are using the speed-perturbed data we need to generate
+# alignments for it.
+# Also the data reverberation will be done in this script/
+echo local/nnet3/multi_condition/run_ivector_common.sh --stage $stage \
+  --clean-data-dir ${clean_train_set} \
+  --iv-dir $iv_dir \
+  --speed-perturb $speed_perturb \
+  --num-data-reps $num_data_reps || exit 1;
+
+
+if [ $stage -le 9 ]; then
+  # Get the alignments as lattices (gives the CTC training more freedom).
+  # use the same num-jobs as the alignments
+  nj=$(cat exp/tri4_ali_nodup${suffix}/num_jobs) || exit 1;
+  steps/align_fmllr_lats.sh --nj $nj --cmd "$train_cmd" data/${clean_train_set}${suffix} \
+    data/lang exp/tri4 $lat_dir
+  rm $lat_dir/fsts.*.gz # save space
+
+
+  # Create the lattices for the reverberated data
+  mkdir -p $rvb_lat_dir/temp/
+  lattice-copy "ark:gunzip -c $lat_dir/lat.*.gz |" ark,scp:$rvb_lat_dir/temp/lats.ark,$rvb_lat_dir/temp/lats.scp
+
+  # copy the lattices for the reverberated data
+  rm -f $rvb_lat_dir/temp/combined_lats.scp
+  touch $rvb_lat_dir/temp/combined_lats.scp
+  # Here prefix "rev0_" represents the clean set, "rev1_" represents the reverberated set
+  for i in `seq 0 $num_data_reps`; do
+    cat $rvb_lat_dir/temp/lats.scp | sed -e "s/^/rev${i}_/" >> $rvb_lat_dir/temp/combined_lats.scp
+  done
+  sort -u $rvb_lat_dir/temp/combined_lats.scp > $rvb_lat_dir/temp/combined_lats_sorted.scp
+
+  lattice-copy scp:$rvb_lat_dir/temp/combined_lats_sorted.scp "ark:|gzip -c >$rvb_lat_dir/lat.1.gz" || exit 1;
+  echo "1" > $rvb_lat_dir/num_jobs
```

Comment:
This looks like it would be extremely slow for large data sets, it's all in one job.
In any case, all the chain-training script (get_egs) does with the lattices is to immediately copy them to ark,scp format, which is what you've done here.  So better to change get_egs.sh so that it requires either lat.*.gz or lat.scp to exist.   If lat.scp exists, all get_egs.sh has to do is copy it to the right directory.